src/main.rs

```diff
@@ -508,17 +508,30 @@ impl Context {
 
             // Construct the path that contains the documentation inside the tarball.
             let tarball_dir = format!("{tarball_prefix}/rustc-docs/share/doc/rust/html");
-            let tarball_dir_new = format!("{tarball_dir}/rustc");
-
-            if Command::new("tar")
-                .arg("tf")
-                .arg(&tarball)
-                .arg(&tarball_dir_new)
-                .current_dir(&rustc_docs)
-                .output()?
-                .status
-                .success()
-            {
+
+            // NOTE: The following logic is to accommodate 3 different `rustc-docs` structures that
+            // have existed over time, in the chronological order:
```

Comment:
I think we should only need to support two formats right now, so presumably one of these can be deleted (probably the oldest one)? This code only needs to support handling artifacts produced by current stable, beta, and main branches in rust-lang/rust, historical releases don't need to be supported.

Reply:
> I think we should only need to support two formats right now [..] This code only needs to support handling artifacts produced by current stable, beta, and main branches [..]

@Mark-Simulacrum If that's the case, no problem!

I've changed the code accordingly, together with the relevant comment/var name adjustments.